packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,301 @@
+    FrontOnly = 2,
+    /** User can only use the back camera */
+    RearOnly = 3,
+  }
+  /**
+   * @hidden
+   * Specifies the image source
+   *
+   * @beta
+   */
+  export enum Source {
+    /** visual media source is camera. */
+    Camera = 1,
+    /** visual media source is gallery. */
+    Gallery = 2,
+  }
+
+  /**
+   * @hidden
+   * VisualMediaFile object that can be used to represent image or video from host apps.
+   *
+   * @beta
+   */
+  export interface VisualMediaFile {
+    /**
+     * @hidden
+     * This is the base64 content of file.
+     * If app needs to use this directly in HTML tags, it should convert this to a data url.
+     */
+    content: string;
+    /**
+     * @hidden
+     * The size of file represented in VisualMediaFile in KB
+     */
+    sizeInKB: number;
+
+    /**
+     * @hidden
+     * Name of the file (does not include the extension)
+     */
+    name: string;
+
+    /**
+     * @hidden
+     * File's MIME type.
+     * Please check https://docs.lens.xyz/docs/metadata-standards#supported-mime-types-for-imagesaudiovideos about more information of mimeType.
+     */
+    mimeType: string;
+  }
+
+  /**
+   * @hidden
+   * Checks whether or not visualMedia has user permission
+   * @returns Promise that will resolve with true if the user had granted the app permission to media information(including Camera and Gallery permission), or with false otherwise,
+   * In case of an error, promise will reject with the error.
+   * @throws NOT_SUPPORTED_ON_PLATFORM Error if the DevicePermission.Media permission has not successfully granted.
+   *
+   * @beta
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.has', permissions);
+  }
+
+  /**
+   * @hidden
+   * Requests user permission for visualMedia
+   * @returns Promise that will resolve with true if the user consented permission for media(including Camera and Gallery permission), or with false otherwise,
+   * In case of an error, promise will reject with the error.
+   * @throws NOT_SUPPORTED_ON_PLATFORM Error if the DevicePermission.Media permission has not successfully granted.
+   *
+   * @beta
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.request', permissions);
+  }
+
+  /**
+   * @hidden
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   *
+   * @beta
+   */
+  export namespace image {
```

Comment:
Can you please explain why we need nested namespace?

Reply:
That's a great question.
In the future, an additional sub-capability named 'video' will be introduced to the visualMedia API (there may be more, but we need to ensure these sub-capabilities are standalone from each other). Thus, the host app has the flexibility to support either the video or the image feature. If the host app only supports one, there's no obligation to implement the other sub-capability.